sound/soc/sof/sof-client-dma-trace.c

```diff
@@ -326,7 +323,8 @@ static ssize_t sof_dfsentry_trace_read(struct file *file, char __user *buffer,
 	}
 
 	/* make sure count is <= avail */
-	count = avail > count ? count : avail;
+	if (count > avail)
+		count = avail;
```

Comment:
should this be added in the initial trace, before moving to the client-based solution?

Reply:
It is not fixing any bugs, but we can consider moving the cleanups up.
It will blow up a bit the conversion patches, so I would rather do that for the final version